client/src/main/java/io/pravega/client/stream/impl/Controller.java

```diff
@@ -87,16 +88,40 @@
 
     /**
      * API to merge or split stream segments.
+     *
+     * @param stream Stream object.
+     * @param sealedSegments List of segments to be sealed.
+     * @param newKeyRanges Key ranges after scaling the stream.
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the scaling was started or not.
+     */
+    CompletableFuture<Boolean> startScale(final Stream stream, final List<Integer> sealedSegments,
+                                           final Map<Double, Double> newKeyRanges);
+
+    /**
+     * API to merge or split stream segments.
```

Comment:
startScale and scaleStream seem to duplicate functionality. It looks to me like either:
a) scaleStream is equivlent to startScale(...).get()
or 
b) This is an attempt to create something that is doubly asynchronous, in which case it would be better to return a future for an object that has APIs related to the ongoing operation. Like getStatus(), cancel(), waitForCompletion()